docs/manual/src/udl/ext_types_wrapped.md

```diff
@@ -57,6 +57,74 @@ Because `UniffiCustomTypeWrapper` is defined in each crate, this means you can w
 if they are not defined in your crate - see the 'wrapper_types' example which demonstrates
 how `serde_json::Value` can be wrapped.
 
+## Error handling during conversion
+
+You might have noticed that the `wrap` function returns a `uniffi::Result<Self>` (which is an
+alias for `anyhow::Result`) and might be wondering what happens if you return an `Err`.
+
+It depends on the context. In short:
+
+* If the value is being used as an argument to a function/constructor that does not return
+  a `Result` (ie, does not have the `throws` attribute in the .udl), then the uniffi generated
+  scaffolding code will `panic!()`
+
+* If the value is being used as an argument to a function/constructor that *does* return a
+  `Result` (ie, does have a `throws` attribute in the .udl), then the uniffi generated
+  scaffolding code will use `anyhow::Error::downcast()` to try and convert the failure into
+  that declared error type and:
+  * If that conversion succeeds, it will be used as the `Err` for the function.
+  * If that conversion fails, it will `panic()`
```

Comment:
I like how this lets us keep using `anyhow::Result` as the return type in all cases.